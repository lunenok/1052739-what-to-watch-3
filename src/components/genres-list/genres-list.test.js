import React from "react";
import renderer from "react-test-renderer";
import GenresList from "../genres-list/genres-list.jsx";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

const uniqueGenres = [`Drama`, `TVshow`];

it(`Should movie card render correctly`, () => {
  const store = mockStore({
    genre: `All genres`,
    currentMovie: null,
    movies: null
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <GenresList uniqueGenres={uniqueGenres}/>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
