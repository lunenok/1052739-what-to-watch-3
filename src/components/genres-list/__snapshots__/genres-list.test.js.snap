// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should movie card render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item + "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item + "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      TVshow
    </a>
  </li>
</ul>
`;
